sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -4,18 +4,572 @@
 #include "azure/core/http/http.hpp"
 #include "azure/core/http/winhttp/win_http_client.hpp"
 
+#include <Windows.h>
+#include <algorithm>
+#include <iostream>
 #include <string>
+#include <winhttp.h>
 
 using namespace Azure::Core::Http;
 
-WinHttpTansport::WinHttpTansport() : HttpTransport() {}
+// WinHttpTransport::WinHttpTransport() : HttpTransport() {}
+//
+// WinHttpTransport::~WinHttpTransport() {}
 
-WinHttpTansport::~WinHttpTansport() {}
+std::wstring HttpMethodToWideString(HttpMethod method)
+{
+  // This string should be all uppercase.
+  // Many servers treat HTTP verbs as case-sensitive, and the Internet Engineering Task Force (IETF)
+  // Requests for Comments (RFCs) spell these verbs using uppercase characters only.
+  switch (method)
+  {
+    case HttpMethod::Get:
+      return L"GET";
+    case HttpMethod::Head:
+      return L"HEAD";
+    case HttpMethod::Post:
+      return L"POST";
+    case HttpMethod::Put:
+      return L"PUT";
+    case HttpMethod::Delete:
+      return L"DELETE";
+    case HttpMethod::Patch:
+      return L"PATCH";
+    default:
+      throw Azure::Core::Http::TransportException("Invalid or unsupported HTTP method.");
+  }
+}
+
+// Convert a UTF-8 string to a wide Unicode string/.
+std::wstring StringToWideString(const std::string& str)
+{
+  int strLength = static_cast<int>(str.size());
+  int sizeNeeded = MultiByteToWideChar(CP_UTF8, 0, str.c_str(), strLength, 0, 0);
```

Comment:
> Or can it happen that for an string "123", the required size for wstring is not 3? (for some encoding stuff?)

Right

> In case the size of the wstring is always the same as the source str, you can do something like

It won't be.